themes/default/content/blog/kubernetes-architecture-templates/index.md

```diff
@@ -0,0 +1,377 @@
+    kubeconfig  : output<string>
+    networkId   : output<string>
+    networkName : "gke-network-e0ce7fb"
+
+Resources:
+    + 6 to create
+
+Do you want to perform this update? yes
+Updating (dev)
+
+View Live: https://app.pulumi.com/nimbinatus/k8s-combo-infra/dev/updates/109
+
+     Type                              Name                 Status
+ +   pulumi:pulumi:Stack               k8s-combo-infra-dev  created (915s)
+ +   ├─ gcp:compute:Network            gke-network          created (21s)
+ +   ├─ gcp:compute:Subnetwork         gke-subnet           created (11s)
+ +   ├─ gcp:container:Cluster          gke-cluster          created (682s)
+ +   ├─ pulumi:providers:kubernetes    app_provider         created (0.17s)
+ +   ├─ gcp:serviceAccount:Account     gke-nodepool-sa      created (1s)
+ +   ├─ kubernetes:core/v1:Namespace   webserver            created (0.52s)
+ +   ├─ gcp:container:NodePool         gke-nodepool         created (125s)
+ +   ├─ kubernetes:core/v1:ConfigMap   webserverconfig      created (0.32s)
+ +   ├─ kubernetes:core/v1:Service     webserverservice     created (193s)
+ +   └─ kubernetes:apps/v1:Deployment  webserverdeployment  created (144s)
+
+
+Outputs:
+    clusterId   : "projects/pulumi-development/locations/us-central1/clusters/gke-cluster-6cc9507"
+    clusterName : "gke-cluster-6cc9507"
+    endpoint_url: "35.224.177.197"
+    kubeconfig  : [secret]
+    networkId   : "projects/pulumi-development/global/networks/gke-network-68b9493"
+    networkName : "gke-network-68b9493"
+
+Resources:
+    + 11 created
+
+Duration: 15m17s
+```
+
+If you `CURL` the endpoint, you get the default NGINX page that tells us the webserver is ready to go:
+
+```bash
+$ curl $(pulumi stack output endpoint_url)
+<!DOCTYPE html>
+<html>
+<head>
+<title>Welcome to nginx!</title>
+<style>
+html { color-scheme: light dark; }
+body { width: 35em; margin: 0 auto;
+font-family: Tahoma, Verdana, Arial, sans-serif; }
+</style>
+</head>
+<body>
+<h1>Welcome to nginx!</h1>
+<p>If you see this page, the nginx web server is successfully installed and
+working. Further configuration is required.</p>
+
+<p>For online documentation and support please refer to
+<a href="http://nginx.org/">nginx.org</a>.<br/>
+Commercial support is available at
+<a href="http://nginx.com/">nginx.com</a>.</p>
+
+<p><em>Thank you for using nginx.</em></p>
+</body>
+</html>
+```
+
+Finally, if you're done with the cluster, don't forget to pull it down so you don't get charged!
+
+```bash
+$ poetry run pulumi destroy -y
+```
+
+If you run into an issue where the cluster gets deleted before the various app resources (the namespace, the service, etc.) get removed, you may get an error that you can't finish destroying the stack. Because the cluster is already gone, the Kubernetes provider can't access it to update the status of the various app resources. So you'll need to remove those resources from your state file as they were removed with the deletion of the cluster:
```

Comment:
I did indeed hit this (I think) -- here's what I got:

```
...
Destroying (dev)

View Live: https://app.pulumi.com/cnunciato/k8s-combo-infra/dev/updates/3

     Type                              Name                 Status                  Info
     pulumi:pulumi:Stack               k8s-combo-infra-dev  **failed**              1 error
 -   ├─ gcp:container:NodePool         gke-nodepool         deleted (244s)          
 -   ├─ gcp:serviceAccount:Account     gke-nodepool-sa      deleted (0.59s)         
 -   ├─ kubernetes:apps/v1:Deployment  webserverdeployment  deleted (1s)            
 -   ├─ kubernetes:core/v1:ConfigMap   webserverconfig      deleted (0.42s)         
 -   ├─ kubernetes:core/v1:Service     webserverservice     deleted (17s)           
 -   ├─ gcp:container:Cluster          gke-cluster          deleted (295s)          
 -   ├─ gcp:compute:Subnetwork         gke-subnet           deleted (11s)           
 -   └─ kubernetes:core/v1:Namespace   webserver            **deleting failed**     1 error

Diagnostics:
  pulumi:pulumi:Stack (k8s-combo-infra-dev):
    error: update failed

  kubernetes:core/v1:Namespace (webserver):
    error: Delete "https://34.66.64.92/api/v1/namespaces/fake-ns": dial tcp 34.66.64.92:443: i/o timeout
    If the cluster has been deleted, you can edit the pulumi state to remove this resource
```

The provided `state delete` command worked, though I had to change the project name in the URN to `k8s-combo-infra` first.

Reply:
Yeah, I don't know if this is a bug or an issue with the dependency tracking layout since it is not done as separate stacks. A lot of people who don't know Pulumi are going to try things this way, though; people like having it run from the root of the repo while having various pieces of code in their own directories. I've seen it asked a number of times in the Slack regarding how to separate out bits of infrastructure while having it all run together. It makes sense, from a new user standpoint, to have everything get stood up together (though to be honest, they're more likely to copy both sets of code into a new file instead of stringing things together like this if they hit this problem).